### ISSUE TITLE
Custom Maps not loading

### ISSUE THREAD

**Issue by vsynth21** [InvoxiPlayGames/BO3MacFix#12]
I am using a 10 core M1 Pro with 32GB of RAM running on Sonoma 14.6.1 and running the game off of an external Samsungs T7 Touch SSD. I have installed, uninstalled and installed wine-stable, as well as XQuartz. The window Says Call of Duty: Black Ops III (with BO3MacFix), but I'm still getting the failed to convert map to Mac Navmesh format on maps like Town Remastered and Der Riese: Declassified. The program no longer boots up a second BO3MacFix console window like it did before, but upon using the terminal to startup the cod2map64.exe with Wine, I get the following output with the same Failed to convert map error on BO3:
Vikrant@Macbook-Pro-140 ~ % wine /Volumes/Vikrant\'s\ 2TB\ Memory\ Card/SteamLibrary/steamapps/common/Call\ of\ Duty\ Black\ Ops\ III/BO3MacFix/cod2map/cod2map64.exe

0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:hid:handle_IRP_MN_QUERY_ID Unhandled type 00000005
0084:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
0084:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
0084:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
0084:fixme:xinput:pdo_pnp IRP_MN_QUERY_ID type 5, not implemented!
DBG: Loaded c2mpatch!
DBG: Installed hooks!
0024:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 00006FFFFDDB7F93, 62) stub
main: Enter
FPU Precision set to 64 bits
CoD2Map: T7
 Build Changelist: 3421987
 Build Machine: CODBUILD8-764
USAGE: cod2map [options] mapname, where options are 0 or more of the following.
Options ignore capitalization; it is only present in the list for clarity.
-platform            Required if not copying a bsp between platforms; specifies 
target platform
-pcToXenon           Copies a PC bsp to a Xenon bsp
-xenonToPc           Copies a Xenon bsp to a PC bsp
-v                   Verbose; enables extra compilation messages
-verboseEntities     Includes verbose messages for submodels if '-v' is given
-onlyEnts            Compile doesn't touch triggers, geometry, or lighting
-generateMapEntsFile Generate map entity string file only, ignore reference pref
abs
-blockSize           Grid size for regular BSP splits; 0 uses largest possible
-subdivisions        Divides all geometry on a grid; only works for small maps
-noSubdivide         Ignores the 'tessSize' setting in all materials
-staticModelCollMaps (Legacy support) Use collmaps for static models
-displayCollMapWarnings Display missing collmap warnings
-ignoreModelErrors   Don't error on missing models, just allow them to drop out
-smoothAngle         Smooth surfaces are only smoothed at angles less than this
-loadFrom            Reads this map instead, but still writes to <mapfile>
-noCurves            Ignores all patch and terrain geometry
-noDetail            Ignores all detail brushes
-fullDetail          Turns all detail brushes into structural brushes
-leakTest            Quits immediately if the map leaked
-portalTest          Forces all portal errors to be fatal
-brushMethod         Brush optimization method (players/bullets/all/none)
-expandPlayer        Writes a map for Radiant to see player-to-brush collision
-expandBullet        Writes a map for Radiant to see bullet-to-brush collision
-debugPortals        Writes a _portals.map showing portal/structural geometry
-noReorderTris       Disables reordering of optimized triangles for T&L cache
-listSlowEntities    Lists entities that process in more than this many seconds
-warnLayerUses       Generates warnings for layer combos used this many or fewer
 times
-warnLayerArea       Generates warnings for layer combos used less than this man
y square inches
-navmesh             Generates the Havok Navigation Mesh during compile
-navvolume           Generates the Havok Navigation Volume during compile. Must 
be generated together with the NavMesh.
-hkdmp               Creates .hkdmp file of navmesh and navvolume to help diagno
se generatio problems.
-pruneNavMesh        Remove unused navmesh faces that don't exist inside any nav
mesh triggers.
-umbraNumThreads     Sets the number of CPU threads Umbra will use when creating
 tome
-umbraSmallestOccluder Surfaces smaller than this value will not be used as occl
uders
-umbraSmallestHole   Geometry gaps smaller than this value are considered solid
-umbraTileSize       Umbra data block size - smaller values will create larger t
omes
-umbraBackfaceLimit  Reverse percentage of backfaces tested to remove bad detail
-umbraObjectGroupCost Groups surfaces together.  Will improve in-game speed but 
degrades occlusion
-umbraEnableSndbs    Enable use of SNDBS to accelerate Umbra compile
-useUnstable         Use unstable shaders & techsetdefs
-spawnTest           Quits immediately if the map has spawn points which interse
ct with level geo
-timing              Enable prints for timing and instrumentation
-prefabInstancing    Prefab instancing behavior [default|always|never]
-noCoalesceCoincidentWindings Turn off coalescing of coincident/coplanar surface
s
-outsource           Compile map for outsource art usage
0024:fixme:msvcrt:__clean_type_info_names_internal (00006FFFFDA91BC8) stub
0024:fixme:msvcrt:__clean_type_info_names_internal (000000007BB8B5F8) stub
0024:fixme:msvcrt:__clean_type_info_names_internal (00006FFFFDEDE030) stub
0024:fixme:msvcrt:__clean_type_info_names_internal (00006FFFFDFE27D8) stub
0024:fixme:msvcrt:__clean_type_info_names_internal (000000007BDEA6C8) stub
0024:fixme:msvcrt:__clean_type_info_names_internal (000000007BFD00D8) stub

**Comment by FillItWithWok** [comment]
Call of the Dead and Der Riese Declassified don't load for me

**Comment by mlane48** [comment]
Are you installing wine-stable through Homebrew? I've gotten the same Mac navmesh convert error before, had to go through some hoops deep in the macOS files to fix it, and was able to install wine-stable through homebrew and it started to work after that.

**Comment by qtSlinky** [comment]
how did you get them to work exactly?

**Comment by vsynth21** [comment]
Turns out, I made mistakes in both my Homebrew installation and my Wine installation. For Brew,  I had to download Xcode command line tools using the commands "sudo rm -rf /Library/Developer/CommandLineTools" and "xcode-select --install". I then reinstalled Brew using the command on the website's home page, but I realized I forgot to set /opt/homebrew/bin to PATH, so I did that by copy-pasting the commands under the warning shortly before the installation complete message. There's a video guide I followed for this part: https://www.youtube.com/watch?v=IWJKRmFLn-g. After that, I used the brew command in the guide to install wine-stable, and it turns out, there was a part of wine I forgot to install too. Wine, too, provides a warning before the "All done!" message that you are not actually done yet, and under it, provides the command "brew install --cask gstreamer-development" to install the GStreamer framework. Finally, after installing both of these, custom maps started to load up.

tl;dr// if you get the "Failed to convert map to Mac Navmesh format" error, your install might be messed up/incomplete. try installing Xcode command line tools and the GStreamer framework through Terminal and update Wine.
